js/src/components/contact-information/phone-number-card/verify-phone-number-content.js

```diff
@@ -0,0 +1,236 @@
+			'Resend code (in %ds)',
+			'google-listings-and-ads'
+		),
+		textSwitch: __(
+			'Or, receive a verification code through a phone call',
+			'google-listings-and-ads'
+		),
+	},
+	[ VERIFICATION_METHOD_PHONE_CALL ]: {
+		toInstruction( phoneNumber ) {
+			return createInterpolateElement(
+				__(
+					'You will receive a phone call at <userPhoneNumber /> with an automated message containing the 6-digit verification code.',
+					'google-listings-and-ads'
+				),
+				{ userPhoneNumber: <strong>{ phoneNumber }</strong> }
+			);
+		},
+		textResend: __( 'Call again', 'google-listings-and-ads' ),
+		// translators: %d: seconds to wait until the next verification code can be requested via phone call.
+		textResendCooldown: __(
+			'Call again (in %ds)',
+			'google-listings-and-ads'
+		),
+		textSwitch: __(
+			'Or, receive a verification code through text message',
+			'google-listings-and-ads'
+		),
+	},
+};
+
+/**
+ * Renders inputs for verifying phone number in Card.Body UI.
+ * Please note that this component will send a verification code request if the current method hasn't been requested yet after rendering.
+ *
+ * @param {Object} props React props.
+ * @param {string} props.verificationMethod The initial verification method.
+ * @param {string} props.country The country code. Example: 'US'.
+ * @param {string} props.number The phone number string in E.164 format. Example: '+12133734253'.
+ * @param {string} props.display The phone number string in international format. Example: '+1 213 373 4253'.
+ * @param {Function} props.onPhoneNumberVerified Called when the phone number is verified.
+ */
+export default function VerifyPhoneNumberContent( {
+	verificationMethod,
+	country,
+	number,
+	display,
+	onPhoneNumberVerified,
+} ) {
+	const [ method, setMethod ] = useState( verificationMethod );
+	const { second, callCount, startCountdown } = useCountdown( method );
+	const [ verification, setVerification ] = useState( null );
+	const [ verifying, setVerifying ] = useState( false );
+	const [ error, setError ] = useState( null );
+	const verificationIdRef = useRef( {} );
+
+	const isSMS = method === VERIFICATION_METHOD_SMS;
+
+	const switchMethod = () => {
+		if ( isSMS ) {
+			setMethod( VERIFICATION_METHOD_PHONE_CALL );
+		} else {
+			setMethod( VERIFICATION_METHOD_SMS );
+		}
+	};
+
+	const handleVerificationCodeRequest = useCallback( () => {
+		setError( null );
+		startCountdown( 60 );
+
+		requestVerificationCode( country, number, method )
+			.then( ( id ) => {
+				verificationIdRef.current[ method ] = id;
+			} )
+			.catch( () => {
+				startCountdown( 0 );
+				// TODO: [full-contact-info] add error handling.
+			} );
+	}, [ country, number, method, startCountdown ] );
+
+	const handleVerifyClick = () => {
+		setError( null );
+		setVerifying( true );
+
+		const id = verificationIdRef.current[ method ];
+		verifyPhoneNumber( id, verification.code, method )
+			.then( () => {
+				onPhoneNumberVerified();
+			} )
```

Comment:
:nail_care: 
Why not just 
```suggestion
			.then( onPhoneNumberVerified )
```

Reply:
:grey_question: BTW, How would that behavior when we re-render the component in the middle of promise resolution?

Wouldn't we call `onPhoneNumberVerified` from the unmounted component, and render the non-verified state?

Reply:
Since the API calling function `verifyPhoneNumber` is not yet implemented, the additional anonymous function is added to prevent its returned value from being passed as unintended parameters to `onPhoneNumberVerified` via the Promise chain.

> How would that behavior when we re-render the component in the middle of promise resolution?

The countdown timer continues to trigger re-render during the middle of promise resolution if you start verifying before the countdown time up. The component would keep being rendered according to its state and props. 

> Wouldn't we call `onPhoneNumberVerified` from the unmounted component, and render the non-verified state?

Not quite sure your meaning. Even if the `onPhoneNumberVerified` is called from an unmounted component, it should still result in a verified phone number card as below. This part will be added by a subsequent PR.

![image](https://user-images.githubusercontent.com/17420811/132450782-aeca2cf1-60b9-4edf-8e81-2dd366990572.png)

Reply:
> Not quite sure your meaning. Even if the onPhoneNumberVerified is called from an unmounted component, it should still result in a verified phone number card as below. This part will be added by a subsequent PR.

What I mean is, more-or-less:
1. Call `VerifyPhoneNumberContent( { onPhoneNumberVerified: A } )`
2. Click "Verify…" (`VerifyPhoneNumberContent~handleVerifyClick`=>`verifyPhoneNumber()`)
3. Call `VerifyPhoneNumberContent( { onPhoneNumberVerified: B } )`
4. Promise from 2. resolves, calling `A()`

Reply:
Do you mean something like this https://codesandbox.io/s/sad-khayyam-5zpus?file=/src/App.js ?

1. Click the button to chose Buzz or Woody - set the `A` callback
2. Click "Greet!"
3. Click the button opposite to the one picked from 1. - set the `B` callback
4. The resolved Promise from 2. calls `A()`
5. The welcomed person and the actual greeted person might be different. Is this your concern?
    ![Kapture 2021-09-08 at 20 12 06](https://user-images.githubusercontent.com/17420811/132507130-097a3704-7915-40ad-a6fb-9040517fc3d9.gif)

Reply:
> `fetchSomebo` 

:rofl: :heart_eyes: 

-----


Yes, something like this. but, in our case when we

Reply:
…in our case, thet's `<Greeting>` button that is about to be re-rendered during promise resolution: https://codesandbox.io/s/nifty-pateu-2zi5q?file=/src/App.js

Try:
1. Click "Buzz…"
2. Click "Greet!"
3. You should see "Verifying…" as expected
4. Before it's resolved, click "Sherif …"
5. Click "Greet!" again.
6. You should see "Verifying…" as expected, before "Howdy, Buzz …" is rendered.
7. :red_square:  but then "Verifying…" is gone, even though, there's still verifying of "Sherif" being done in the background.

![verifying](https://user-images.githubusercontent.com/17435/132677680-c48c1de3-5009-4870-8201-6451764468cd.gif)

Reply:
I think I managed to reproduce my hypothetical problem at https://github.com/woocommerce/google-listings-and-ads/issues/1004

Reply:
The problem in 7. would become a concern when there's a way to trigger another same task of promise during the previous one is not yet fulfilled, such as step 5. But this case should not happen in the `<VerifyPhoneNumberContent>`.

And issue #1004 is more about updating state on an unmounted component, like the `verifying`, `error`, and countdown states within the `<VerifyPhoneNumberContent>`. It's an issue but I think it doesn't relate to the verified state. The verified state will be handled in the upper scope or the wp-data store. Even though a user navigates away during an issued promise processing, these unexpected cases could be handled by the consumer component of `<VerifyPhoneNumberContent>` when the `onPhoneNumberVerified` calling back.